src/retail_calendar.ts

```diff
@@ -41,6 +43,32 @@ export const RetailCalendarFactory: RetailCalendarConstructor = class Calendar
       .startOf('day')
     this.weeks = this.generateWeeks()
     this.months = this.generateMonths()
+    this.leapYearStrategy = this.getLeapYearStrategy()
```

Comment:
I think this should be moved to immediately after line #37. We need this.leapYearStrategy to be set, before we can calculate rest of the stuff.